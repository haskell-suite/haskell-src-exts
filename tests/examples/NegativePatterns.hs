f 1 = -1
f (-1) = 1
f ( - 2) = 2
f (  -  3) = 3

data Z a = Higher a a | Same a a | Lower a a

infixr 7 `Higher`
infixr 6 `Same`
infixr 5 `Lower`

g :: Z Int -> ()
g (   -1 `Higher`  x) = ()
g (  -  2 `Same`   x) = ()
g ( -    3 `Lower` x) = ()
