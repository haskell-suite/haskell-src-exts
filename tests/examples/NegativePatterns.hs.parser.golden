ParseOk (Module (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 1, srcSpanStartColumn = 1, srcSpanEndLine = 5, srcSpanEndColumn = 1}, srcInfoPoints = [SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 1, srcSpanStartColumn = 1, srcSpanEndLine = 1, srcSpanEndColumn = 1},SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 1, srcSpanStartColumn = 1, srcSpanEndLine = 1, srcSpanEndColumn = 1},SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 1, srcSpanStartColumn = 1, srcSpanEndLine = 1, srcSpanEndColumn = 1},SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 2, srcSpanStartColumn = 1, srcSpanEndLine = 2, srcSpanEndColumn = 1},SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 3, srcSpanStartColumn = 1, srcSpanEndLine = 3, srcSpanEndColumn = 1},SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 4, srcSpanStartColumn = 1, srcSpanEndLine = 4, srcSpanEndColumn = 1},SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 5, srcSpanStartColumn = 1, srcSpanEndLine = 5, srcSpanEndColumn = 1},SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 5, srcSpanStartColumn = 1, srcSpanEndLine = 5, srcSpanEndColumn = 1}]}) Nothing [] [] [FunBind (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 1, srcSpanStartColumn = 1, srcSpanEndLine = 4, srcSpanEndColumn = 15}, srcInfoPoints = []}) [Match (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 1, srcSpanStartColumn = 1, srcSpanEndLine = 1, srcSpanEndColumn = 9}, srcInfoPoints = []}) (Ident (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 1, srcSpanStartColumn = 1, srcSpanEndLine = 1, srcSpanEndColumn = 2}, srcInfoPoints = []}) "f") [PLit (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 1, srcSpanStartColumn = 3, srcSpanEndLine = 1, srcSpanEndColumn = 4}, srcInfoPoints = []}) (Positive (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 1, srcSpanStartColumn = 3, srcSpanEndLine = 1, srcSpanEndColumn = 4}, srcInfoPoints = []})) (Int (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 1, srcSpanStartColumn = 3, srcSpanEndLine = 1, srcSpanEndColumn = 4}, srcInfoPoints = []}) 1 "1")] (UnGuardedRhs (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 1, srcSpanStartColumn = 5, srcSpanEndLine = 1, srcSpanEndColumn = 9}, srcInfoPoints = [SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 1, srcSpanStartColumn = 5, srcSpanEndLine = 1, srcSpanEndColumn = 6}]}) (NegApp (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 1, srcSpanStartColumn = 7, srcSpanEndLine = 1, srcSpanEndColumn = 9}, srcInfoPoints = [SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 1, srcSpanStartColumn = 7, srcSpanEndLine = 1, srcSpanEndColumn = 8}]}) (Lit (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 1, srcSpanStartColumn = 8, srcSpanEndLine = 1, srcSpanEndColumn = 9}, srcInfoPoints = []}) (Int (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 1, srcSpanStartColumn = 8, srcSpanEndLine = 1, srcSpanEndColumn = 9}, srcInfoPoints = []}) 1 "1")))) Nothing,Match (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 2, srcSpanStartColumn = 1, srcSpanEndLine = 2, srcSpanEndColumn = 11}, srcInfoPoints = []}) (Ident (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 2, srcSpanStartColumn = 1, srcSpanEndLine = 2, srcSpanEndColumn = 2}, srcInfoPoints = []}) "f") [PParen (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 2, srcSpanStartColumn = 3, srcSpanEndLine = 2, srcSpanEndColumn = 7}, srcInfoPoints = [SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 2, srcSpanStartColumn = 3, srcSpanEndLine = 2, srcSpanEndColumn = 4},SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 2, srcSpanStartColumn = 6, srcSpanEndLine = 2, srcSpanEndColumn = 7}]}) (PLit (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 2, srcSpanStartColumn = 4, srcSpanEndLine = 2, srcSpanEndColumn = 6}, srcInfoPoints = [SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 2, srcSpanStartColumn = 4, srcSpanEndLine = 2, srcSpanEndColumn = 5}]}) (Negative (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 2, srcSpanStartColumn = 4, srcSpanEndLine = 2, srcSpanEndColumn = 5}, srcInfoPoints = [SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 2, srcSpanStartColumn = 4, srcSpanEndLine = 2, srcSpanEndColumn = 5}]})) (Int (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 2, srcSpanStartColumn = 5, srcSpanEndLine = 2, srcSpanEndColumn = 6}, srcInfoPoints = []}) 1 "1"))] (UnGuardedRhs (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 2, srcSpanStartColumn = 8, srcSpanEndLine = 2, srcSpanEndColumn = 11}, srcInfoPoints = [SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 2, srcSpanStartColumn = 8, srcSpanEndLine = 2, srcSpanEndColumn = 9}]}) (Lit (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 2, srcSpanStartColumn = 10, srcSpanEndLine = 2, srcSpanEndColumn = 11}, srcInfoPoints = []}) (Int (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 2, srcSpanStartColumn = 10, srcSpanEndLine = 2, srcSpanEndColumn = 11}, srcInfoPoints = []}) 1 "1"))) Nothing,Match (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 3, srcSpanStartColumn = 1, srcSpanEndLine = 3, srcSpanEndColumn = 13}, srcInfoPoints = []}) (Ident (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 3, srcSpanStartColumn = 1, srcSpanEndLine = 3, srcSpanEndColumn = 2}, srcInfoPoints = []}) "f") [PParen (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 3, srcSpanStartColumn = 3, srcSpanEndLine = 3, srcSpanEndColumn = 9}, srcInfoPoints = [SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 3, srcSpanStartColumn = 3, srcSpanEndLine = 3, srcSpanEndColumn = 4},SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 3, srcSpanStartColumn = 8, srcSpanEndLine = 3, srcSpanEndColumn = 9}]}) (PLit (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 3, srcSpanStartColumn = 5, srcSpanEndLine = 3, srcSpanEndColumn = 8}, srcInfoPoints = [SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 3, srcSpanStartColumn = 5, srcSpanEndLine = 3, srcSpanEndColumn = 6}]}) (Negative (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 3, srcSpanStartColumn = 5, srcSpanEndLine = 3, srcSpanEndColumn = 6}, srcInfoPoints = [SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 3, srcSpanStartColumn = 5, srcSpanEndLine = 3, srcSpanEndColumn = 6}]})) (Int (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 3, srcSpanStartColumn = 7, srcSpanEndLine = 3, srcSpanEndColumn = 8}, srcInfoPoints = []}) 2 "2"))] (UnGuardedRhs (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 3, srcSpanStartColumn = 10, srcSpanEndLine = 3, srcSpanEndColumn = 13}, srcInfoPoints = [SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 3, srcSpanStartColumn = 10, srcSpanEndLine = 3, srcSpanEndColumn = 11}]}) (Lit (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 3, srcSpanStartColumn = 12, srcSpanEndLine = 3, srcSpanEndColumn = 13}, srcInfoPoints = []}) (Int (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 3, srcSpanStartColumn = 12, srcSpanEndLine = 3, srcSpanEndColumn = 13}, srcInfoPoints = []}) 2 "2"))) Nothing,Match (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 4, srcSpanStartColumn = 1, srcSpanEndLine = 4, srcSpanEndColumn = 15}, srcInfoPoints = []}) (Ident (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 4, srcSpanStartColumn = 1, srcSpanEndLine = 4, srcSpanEndColumn = 2}, srcInfoPoints = []}) "f") [PParen (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 4, srcSpanStartColumn = 3, srcSpanEndLine = 4, srcSpanEndColumn = 11}, srcInfoPoints = [SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 4, srcSpanStartColumn = 3, srcSpanEndLine = 4, srcSpanEndColumn = 4},SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 4, srcSpanStartColumn = 10, srcSpanEndLine = 4, srcSpanEndColumn = 11}]}) (PLit (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 4, srcSpanStartColumn = 6, srcSpanEndLine = 4, srcSpanEndColumn = 10}, srcInfoPoints = [SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 4, srcSpanStartColumn = 6, srcSpanEndLine = 4, srcSpanEndColumn = 7}]}) (Negative (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 4, srcSpanStartColumn = 6, srcSpanEndLine = 4, srcSpanEndColumn = 7}, srcInfoPoints = [SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 4, srcSpanStartColumn = 6, srcSpanEndLine = 4, srcSpanEndColumn = 7}]})) (Int (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 4, srcSpanStartColumn = 9, srcSpanEndLine = 4, srcSpanEndColumn = 10}, srcInfoPoints = []}) 3 "3"))] (UnGuardedRhs (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 4, srcSpanStartColumn = 12, srcSpanEndLine = 4, srcSpanEndColumn = 15}, srcInfoPoints = [SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 4, srcSpanStartColumn = 12, srcSpanEndLine = 4, srcSpanEndColumn = 13}]}) (Lit (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 4, srcSpanStartColumn = 14, srcSpanEndLine = 4, srcSpanEndColumn = 15}, srcInfoPoints = []}) (Int (SrcSpanInfo {srcInfoSpan = SrcSpan {srcSpanFilename = "tests/examples/NegativePatterns.hs", srcSpanStartLine = 4, srcSpanStartColumn = 14, srcSpanEndLine = 4, srcSpanEndColumn = 15}, srcInfoPoints = []}) 3 "3"))) Nothing]],[])
